File: Deepspace_Ruby/P1/SuppliesPackage.rb
# BOLÍVAR PELÁEZ, CLARA
# RUIZ GÓMEZ, SOLEDAD
# 2º DGIIM - PDOO - Practice 1 DeepSpace

# SuppliesPackage
# It represents a spacial station's supplies

module DeepSpace

    class SuppliesPackage 

        # Constructor
        def initialize (a, f, s)
            @ammoPower      = a
            @fuelUnits      = f 
            @shieldPower    = s
        end

        # Copy constructor
        def self.newCopy (s)
            new(s.ammoPower, s.fuelUnits, s.shieldPower)
        end

        # Getters
        def ammoPower
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end

        def shieldPower
            @shieldPower
        end

    end

end

File: Deepspace_Java/P4/src/deepspace/SuppliesPackage.java
/**
 * @author BOLÍVAR PELÁEZ, CLARA
 * @author RUIZ GÓMEZ, SOLEDAD
 * 2º DGIIM - PDOO - Practice 1 DeepSpace
 */
package deepspace;

/**
 * @title SuppliesPackage
 * @brief It represents a spacial station's supplies
 */
public class SuppliesPackage {
    
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    /**
    * @brief Constructor
    */
    SuppliesPackage (float a, float f, float s) {
        ammoPower   = a;
        fuelUnits   = f;
        shieldPower = s;
    }
    
    /**
    * @brief Copy constructor
    */
    SuppliesPackage (SuppliesPackage s) {
        ammoPower   = s.ammoPower;
        fuelUnits   = s.fuelUnits;
        shieldPower = s.shieldPower;
    }
    
    /**
    * @brief Getter with package visibility
    * @return ammoPower value
    */
    float getAmmoPower () {
        return ammoPower;
    }
    
    /**
    * @brief Getter with package visibility
    * @return fuelUnits value
    */
    float getFuelUnits () {
        return fuelUnits;
    }
    
    /**
    * @brief Getter with package visibility
    * @return shieldPower value
    */
    float getShieldPower () {
        return shieldPower;
    }
    
    /**
    * @brief toString
    * @return String with info about the instance
    */
    @Override
    public String toString () {
        
        String output = "SuppliesPackage [ ammoPower " + ammoPower 
                        + " ; fuelUnits " + fuelUnits 
                        +  " ; shieldPower " + shieldPower +" ]";
        
        return output;
    }
    
 
}
